staging/src/k8s.io/cloud-provider/node/helpers/address.go

```diff
@@ -88,14 +88,12 @@ func GetNodeAddressesFromNodeIPLegacy(nodeIP net.IP, cloudNodeAddresses []v1.Nod
 	return GetNodeAddressesFromNodeIP(nodeIP.String(), cloudNodeAddresses)
 }
 
-// GetNodeAddressesFromNodeIP filters the provided list of nodeAddresses to match the
-// providedNodeIP from the Node annotation (which is assumed to be non-empty). This is
+// GetNodeAddressesFromNodeIP sorts the provided list of nodeAddresses to use the
```

Comment:
if you're going to change the behavior here, you should probably just sync it to the behavior of `GetNodeAddressesFromNodeIPLegacy`